packages/api/repository/PortfolioRepository.ts

```diff
@@ -0,0 +1,55 @@
+import { EntityRepository, Repository, InsertResult, UpdateResult } from "typeorm";
+import { IPortfolio, IPortfolioCreate, Portfolio } from "../../orm/entity/Portfolio";
+
+@EntityRepository(Portfolio)
+export class PortfolioRepository extends Repository<Portfolio> {
+  // GET portfolios/:id
+  async getPortfolio(id: string): Promise<Portfolio> {
+    return await this.findOneOrFail({
+      select: [
+        "id",
+        "name",
+        "description",
+        "owner",
+        "csp",
+        "dodComponents",
+        "portfolioManagers",
+        "createdAt",
+        "updatedAt",
+        "archivedAt",
+        "administrators",
+      ],
+      where: { id },
+    });
+  }
+
+  // GET all portfolios
+  async getPortfolios(): Promise<Array<Portfolio>> {
+    return await this.createQueryBuilder("portfolio")
+      .select(["portfolio.name", "portfolio.id", "portfolio.createdAt", "portfolio.updatedAt", "portfolio.archivedAt"])
+      .getMany();
+  }
+
+  // POST create new portfolio
+  async createPortfolio(portfolio: IPortfolio): Promise<InsertResult> {
+    return await this.insert(portfolio);
```

Comment:
I know you mentioned about seeing the updates display old data previously. Did you see a difference once the `async/await` was added? 

I know previously I added `async/await` on all the methods, but sonar cloud complained about a lot of them.

Reply:
I've not looked at this since then.  Will check sonar cloud and follow guidance there.

Reply:
Corrected smells in 8b9aca2fa37b114313e9bd7785ffbd4cfbe2eaa4